addon-registry.js

```diff
@@ -472,6 +472,37 @@ class AddonConfigurationRegistry {
   getAddonDependencies() {
     return getAddonsLoaderChain(this.dependencyGraph);
   }
+
+  getDotDependencyGraph() {
+    const seen = [];
```

Comment:
Probably doesn't matter at all for performance at this scale, but it might make sense to use a Set here. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set